geopandas/tools/sjoin.py

```diff
@@ -151,8 +179,20 @@ def _basic_checks(left_df, right_df, how, lsuffix, rsuffix):
     if not _check_crs(left_df, right_df):
         _crs_mismatch_warn(left_df, right_df, stacklevel=4)
 
+    if on_attribute:
+        for attr in on_attribute:
+            if not ((attr in left_df) and (attr in right_df)):
+                raise ValueError(
+                    f"Expected column {attr} is missing from one of the dataframes"
+                )
```

Comment:
Can we enhance the warning a bit to say which dataframe(s) are missing the column?

Reply:
Thanks Martin, have updated to provide that information in the error msg